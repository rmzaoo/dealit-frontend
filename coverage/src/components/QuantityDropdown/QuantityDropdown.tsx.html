
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/QuantityDropdown/QuantityDropdown.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">src/components/QuantityDropdown</a> QuantityDropdown.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>0/85</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>0/85</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a></td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="branch-0 cbranch-no" title="branch not covered" >import React, { useState, useRef, useEffect } from "react";</span></span></span>
<span class="cstat-no" title="statement not covered" >import { MdKeyboardArrowUp, MdKeyboardArrowDown } from "react-icons/md";</span>
<span class="cstat-no" title="statement not covered" >import _ from "lodash";</span>
<span class="cstat-no" title="statement not covered" >import {</span>
<span class="cstat-no" title="statement not covered" >  DropdownAndTextContainer,</span>
<span class="cstat-no" title="statement not covered" >  TextContainer,</span>
<span class="cstat-no" title="statement not covered" >  Text,</span>
<span class="cstat-no" title="statement not covered" >  DropdownContainer,</span>
<span class="cstat-no" title="statement not covered" >  StyledDropdown,</span>
<span class="cstat-no" title="statement not covered" >  DropdownText,</span>
<span class="cstat-no" title="statement not covered" >  DrodownArrowContainer,</span>
<span class="cstat-no" title="statement not covered" >  DropdownContentContainer,</span>
<span class="cstat-no" title="statement not covered" >  DropdownContentOption,</span>
<span class="cstat-no" title="statement not covered" >} from "./style";</span>
<span class="cstat-no" title="statement not covered" ></span>
<span class="cstat-no" title="statement not covered" >const QuantityDropdown = ({ optionSelected, setOptionSelected }: any) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >  const [isOpen, setIsOpen] = useState(false);</span>
<span class="cstat-no" title="statement not covered" >  const wrapperRef = useRef(null);</span>
<span class="cstat-no" title="statement not covered" ></span>
<span class="cstat-no" title="statement not covered" >  const dropdownToggle = () =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    setIsOpen(!isOpen);</span>
<span class="cstat-no" title="statement not covered" >  };</span>
<span class="cstat-no" title="statement not covered" ></span>
<span class="cstat-no" title="statement not covered" >  //DETECT MOUSE CLICKS OUTSIDE</span>
<span class="cstat-no" title="statement not covered" >  const useOutsideAlerter = (ref: any) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    useEffect(() =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      //CLICKED OUTSIDE OF ELEMENT</span>
<span class="cstat-no" title="statement not covered" >      const handler = (event: { target: any }) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        if (isOpen &amp;&amp; ref.current &amp;&amp; !ref.current.contains(event.target)) {</span>
<span class="cstat-no" title="statement not covered" >          setIsOpen(false);</span>
<span class="cstat-no" title="statement not covered" >        }</span>
<span class="cstat-no" title="statement not covered" >      };</span>
<span class="cstat-no" title="statement not covered" ></span>
<span class="cstat-no" title="statement not covered" >      //BINDE EVENT LISTENER</span>
<span class="cstat-no" title="statement not covered" >      document.addEventListener("mousedown", handler);</span>
<span class="cstat-no" title="statement not covered" >      return () =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        // UNBIND EVENT LISTENER ON CLEANUP</span>
<span class="cstat-no" title="statement not covered" >        document.removeEventListener("mousedown", handler);</span>
<span class="cstat-no" title="statement not covered" >      };</span>
<span class="cstat-no" title="statement not covered" >    }, [ref, isOpen]);</span>
<span class="cstat-no" title="statement not covered" >  };</span>
<span class="cstat-no" title="statement not covered" ></span>
<span class="cstat-no" title="statement not covered" >  const getDropdownOptions = (index: number) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
<span class="cstat-no" title="statement not covered" >      &lt;DropdownContentOption</span>
<span class="cstat-no" title="statement not covered" >        onClick={() =&gt; {</span>
<span class="cstat-no" title="statement not covered" >          setOptionSelected(index + 1);</span>
<span class="cstat-no" title="statement not covered" >          dropdownToggle();</span>
<span class="cstat-no" title="statement not covered" >        }}</span>
<span class="cstat-no" title="statement not covered" >        active={index + 1 === optionSelected}</span>
<span class="cstat-no" title="statement not covered" >      &gt;</span>
<span class="cstat-no" title="statement not covered" >        {index + 1}</span>
<span class="cstat-no" title="statement not covered" >      &lt;/DropdownContentOption&gt;</span>
<span class="cstat-no" title="statement not covered" >    );</span>
<span class="cstat-no" title="statement not covered" >  };</span>
<span class="cstat-no" title="statement not covered" ></span>
<span class="cstat-no" title="statement not covered" >  useOutsideAlerter(wrapperRef);</span>
<span class="cstat-no" title="statement not covered" ></span>
<span class="cstat-no" title="statement not covered" >  return (</span>
<span class="cstat-no" title="statement not covered" >    &lt;DropdownAndTextContainer&gt;</span>
<span class="cstat-no" title="statement not covered" >      &lt;TextContainer&gt;</span>
<span class="cstat-no" title="statement not covered" >        &lt;Text&gt;Quantity:&lt;/Text&gt;</span>
<span class="cstat-no" title="statement not covered" >      &lt;/TextContainer&gt;</span>
<span class="cstat-no" title="statement not covered" >      &lt;DropdownContainer ref={wrapperRef}&gt;</span>
<span class="cstat-no" title="statement not covered" >        &lt;StyledDropdown onClick={() =&gt; dropdownToggle()} active={isOpen}&gt;</span>
<span class="cstat-no" title="statement not covered" >          &lt;DropdownText&gt;{optionSelected}&lt;/DropdownText&gt;</span>
<span class="cstat-no" title="statement not covered" >          &lt;DrodownArrowContainer&gt;</span>
<span class="cstat-no" title="statement not covered" >            {isOpen ? (</span>
<span class="cstat-no" title="statement not covered" >              &lt;MdKeyboardArrowUp style={{ verticalAlign: "middle" }} /&gt;</span>
<span class="cstat-no" title="statement not covered" >            ) : (</span>
<span class="cstat-no" title="statement not covered" >              &lt;MdKeyboardArrowDown style={{ verticalAlign: "middle" }} /&gt;</span>
<span class="cstat-no" title="statement not covered" >            )}</span>
<span class="cstat-no" title="statement not covered" >          &lt;/DrodownArrowContainer&gt;</span>
<span class="cstat-no" title="statement not covered" >        &lt;/StyledDropdown&gt;</span>
<span class="cstat-no" title="statement not covered" >        {isOpen &amp;&amp; (</span>
<span class="cstat-no" title="statement not covered" >          &lt;DropdownContentContainer&gt;</span>
<span class="cstat-no" title="statement not covered" >            {_.times(15, getDropdownOptions)}</span>
<span class="cstat-no" title="statement not covered" >          &lt;/DropdownContentContainer&gt;</span>
<span class="cstat-no" title="statement not covered" >        )}</span>
<span class="cstat-no" title="statement not covered" >      &lt;/DropdownContainer&gt;</span>
<span class="cstat-no" title="statement not covered" >    &lt;/DropdownAndTextContainer&gt;</span>
<span class="cstat-no" title="statement not covered" >  );</span>
<span class="cstat-no" title="statement not covered" >};</span>
<span class="cstat-no" title="statement not covered" ></span>
<span class="cstat-no" title="statement not covered" >export default QuantityDropdown;</span>
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Wed Jun 15 2022 12:26:33 GMT+0100 (Hora de verão da Europa Ocidental)
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    