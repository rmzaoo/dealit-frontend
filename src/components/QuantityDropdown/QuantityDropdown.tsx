import React, { useState, useRef, useEffect } from "react";
import { MdKeyboardArrowUp, MdKeyboardArrowDown } from "react-icons/md";
import _ from "lodash";
import {
  DropdownAndTextContainer,
  TextContainer,
  Text,
  DropdownContainer,
  StyledDropdown,
  DropdownText,
  DrodownArrowContainer,
  DropdownContentContainer,
  DropdownContentOption,
} from "./style";

const QuantityDropdown = ({ optionSelected, setOptionSelected }: any) => {
  const [isOpen, setIsOpen] = useState(false);
  const wrapperRef = useRef(null);

  const dropdownToggle = () => {
    setIsOpen(!isOpen);
  };

  //DETECT MOUSE CLICKS OUTSIDE
  const useOutsideAlerter = (ref: any) => {
    useEffect(() => {
      //CLICKED OUTSIDE OF ELEMENT
      const handler = (event: { target: any }) => {
        if (isOpen && ref.current && !ref.current.contains(event.target)) {
          setIsOpen(false);
        }
      };

      //BINDE EVENT LISTENER
      document.addEventListener("mousedown", handler);
      return () => {
        // UNBIND EVENT LISTENER ON CLEANUP
        document.removeEventListener("mousedown", handler);
      };
    }, [ref, isOpen]);
  };

  const getDropdownOptions = (index: number) => {
    return (
      <DropdownContentOption
        onClick={() => {
          setOptionSelected(index + 1);
          dropdownToggle();  
        }}
        active ={index + 1 === optionSelected}
      >
        {index + 1}
      </DropdownContentOption>
    );
  };

  useOutsideAlerter(wrapperRef);

  return (
    <DropdownAndTextContainer>
      <TextContainer>
        <Text>Quantity:</Text>
      </TextContainer>
      <DropdownContainer ref={wrapperRef}>
        <StyledDropdown onClick={() => dropdownToggle()}>
          <DropdownText>{optionSelected}</DropdownText>
          <DrodownArrowContainer>
            {isOpen ? (
              <MdKeyboardArrowUp style={{ verticalAlign: "middle" }} />
            ) : (
              <MdKeyboardArrowDown style={{ verticalAlign: "middle" }} />
            )}
          </DrodownArrowContainer>
        </StyledDropdown>
        {isOpen && (
          <DropdownContentContainer>
            {_.times(15, getDropdownOptions)}
          </DropdownContentContainer>
        )}
      </DropdownContainer>
    </DropdownAndTextContainer>
  );
};

export default QuantityDropdown;
