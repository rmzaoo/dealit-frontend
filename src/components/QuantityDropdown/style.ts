import styled, { keyframes } from "styled-components";

interface Props {
  active?: boolean;
}

const fadeIn = keyframes`
  0% {
        transform: scaleY(0)
    }
    80% {
        transform: scaleY(1.1)
    }
    100% {
        transform: scaleY(1)
    }
`;

export const DropdownAndTextContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  width: 100%;
`;

export const TextContainer = styled.div`
  display: flex;
  justify-content: flex-start;
`;

export const Text = styled.p`
  font-size: 1.2rem;
  color: white;
`;

export const DropdownContainer = styled.div`
  position: relative;
  display: inline-block;
  justify-content: center;
  align-items: center;
  width: 14%;
  margin-left: 10px;
`;

export const StyledDropdown = styled.div<Props>`
  display: flex;
  position: relative;
  justify-content: space-around;
  align-items: center;
  border: ${(props) => (props.active ? "2px solid #00bcd4" : "2px solid white")};
  cursor: pointer;
  height: 27px;
  padding: 0px 5px;
  width: 100%;

  &:hover {
    border: 2px solid #00bcd4;
  }
`;

export const DropdownText = styled.p`
  font-size: 1.1rem;
  color: white;
`;

export const DrodownArrowContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.5rem;
  color: white;
`;

export const DropdownContentContainer = styled.div`
  position: absolute;
  min-width: 100%;
  border-left: 2px solid #00bcd4;
  border-right: 2px solid #00bcd4;
  border-bottom: 2px solid #00bcd4;
  width: 15%;
  overflow: auto;
  z-index: 2;
  transition: all 0.3s ease-out;
  animation: 300ms ease-in-out forwards;
  transform-origin: top center;
  padding: 5px;
  background-color: #e9e9ed;
  animation: ${fadeIn} 0.4s ease-in-out;
  cursor: pointer;
`;

export const DropdownContentOption = styled.button<Props>`
  width: 100%;
  border: 0px 2px 2px 2px solid black;
  border-style: none;
  text-align: center;
  cursor: pointer;
  background-color: ${(props) => (props.active ? "#27364b" : "#e9e9ed")};
  color: ${(props) => (props.active ? "white" : "black")};

  &:hover {
    background-color: ${(props) => (props.active ? "#27364b" : "#cdcdcd")};
  }
`;
