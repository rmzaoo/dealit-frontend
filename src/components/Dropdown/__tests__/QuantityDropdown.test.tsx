import React from "react";
import { describe, test, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import QuantityDropdown from "../Dropdown";

describe("QuantityDropdown", () => {
  test("Renders without crashin", () => {
    const result = render(
      <BrowserRouter>
        <QuantityDropdown />
      </BrowserRouter>
    );

    expect(result).toMatchSnapshot();
  });
});
