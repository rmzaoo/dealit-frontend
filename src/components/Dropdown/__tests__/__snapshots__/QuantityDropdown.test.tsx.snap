// Vitest Snapshot v1

exports[`QuantityDropdown > Renders without crashin 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ gPXlOd"
      >
        <div
          class="sc-gsnTZi evBwtk"
        >
          <p
            class="sc-dkzDqf bhEbba"
          >
            Quantity:
          </p>
        </div>
        <div
          class="sc-hKMtZM fUPfiV"
        >
          <div
            class="sc-eCYdqJ hzsJqb"
          >
            <p
              class="sc-jSMfEi jvVtXX"
            />
            <div
              class="sc-gKXOVf bRKVfn"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="vertical-align: middle;"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ gPXlOd"
    >
      <div
        class="sc-gsnTZi evBwtk"
      >
        <p
          class="sc-dkzDqf bhEbba"
        >
          Quantity:
        </p>
      </div>
      <div
        class="sc-hKMtZM fUPfiV"
      >
        <div
          class="sc-eCYdqJ hzsJqb"
        >
          <p
            class="sc-jSMfEi jvVtXX"
          />
          <div
            class="sc-gKXOVf bRKVfn"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
